src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs

```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
+                        {
+                            return "this";
+                        }
+                        return string.Empty;
                     }
 
-                    if (_parameter.IsParams)
+                    string VB()
                     {
-                        return "params";
+                        switch (ParameterSymbol.RefKind)
+                        {
+                            case RefKind.Ref:
+                                return "ByRef";
+                        }
+
+                        if (ParameterSymbol.IsParams)
+                        {
+                            return "ParamArray";
+                        }
+
+                        if (_changeSignatureDialogViewModel._thisParameter != null &&
+                            ParameterSymbol == _changeSignatureDialogViewModel._thisParameter.ParameterSymbol)
+                        {
+                            return "Me";
+                        }
+                        return string.Empty;
                     }
 
-                    if (_changeSignatureDialogViewModel._thisParameter != null &&
-                        _parameter == _changeSignatureDialogViewModel._thisParameter._parameter)
+                    switch (ParameterSymbol.Language)
                     {
-                        return "this";
+                        case "C#":
+                            return CS();
+                        case "Visual Basic":
+                            return VB();
+                        default:
+                            return string.Empty;
                     }
-
-                    return string.Empty;
                 }
             }
 
-            public string Type
-            {
-                get { return _parameter.Type.ToDisplayString(s_parameterDisplayFormat); }
-            }
+            public string Type => ParameterSymbol.Type.ToDisplayString(s_parameterDisplayFormat);
 
-            public string Parameter
-            {
-                get { return _parameter.Name; }
-            }
+            public string Parameter => ParameterSymbol.Name;
 
             public string Default
             {
                 get
                 {
-                    if (!_parameter.HasExplicitDefaultValue)
+                    string CS()
+                    {
+                        return ParameterSymbol.ExplicitDefaultValue == null
+                        ? "null"
+                        : ParameterSymbol.ExplicitDefaultValue is string
+                            ? "\"" + ParameterSymbol.ExplicitDefaultValue.ToString() + "\""
+                            : ParameterSymbol.ExplicitDefaultValue.ToString();
+                    }
+                    string VB()
+                    {
+                        return ParameterSymbol.ExplicitDefaultValue == null ? "Nothing" :
+                               ParameterSymbol.ExplicitDefaultValue.ToString();
```

Comment:
not following IDE types (shown above) for `?:`.  Why not handle the string case like the C# version does?